html/infrastructure/safe-passing-of-structured-data/structured-cloning-error-stack-optional.window.js

```diff
@@ -0,0 +1,51 @@
+stackTest(() => {
+  return new Error('some message');
+}, 'page-created Error');
+
+stackTest(() => {
+  return new DOMException('InvalidStateError', 'some message');
+}, 'page-created DOMException');
+
+stackTest(() => {
+  try {
+    Object.defineProperty();
+  } catch (e) {
+    return e;
+  }
+}, 'JS-engine-created TypeError');
+
+stackTest(() => {
+  try {
+    HTMLParagraphElement.prototype.align;
+  } catch (e) {
+    return e;
+  }
+}, 'web API-created TypeError');
+
+stackTest(() => {
+  try {
+    document.createElement('');
+  } catch (e) {
+    return e;
+  }
+}, 'web API-created DOMException');
+
+
+function stackTest(errorFactory, description) {
+  async_test(t => {
+    const error = errorFactory();
+    const originalStack = error.stack;
+
+    if (!originalStack) {
+      t.done();
+      return;
+    }
+
+    const worker = new Worker('./echo.js');
```

Comment:
Should we also add a test with SharedWorker so it goes across agent cluster boundaries?

Reply:
Maybe service worker or cross-origin iframe since that has broader browser support. Any objections to that? I guess cross-origin iframe would be easiest.

Reply:
It'd have to be a cross-site frame.